src/Parsers/IParser.h

```diff
@@ -73,6 +74,21 @@ class IParser
             if (unlikely(max_depth > 0 && depth > max_depth))
                 throw Exception(ErrorCodes::TOO_DEEP_RECURSION, "Maximum parse depth ({}) exceeded. "
                     "Consider rising max_parser_depth parameter.", max_depth);
+
+            /** Sometimes the maximum parser depth can be set to a high value by the user,
+              * but we still want to avoid stack overflow.
+              * For this purpose, we can use the checkStackSize function, but it is too heavy.
+              * The solution is to check not too frequently.
+              * The frequency is arbitrary, but not too large, not too small,
+              * and a power of two to simplify the division.
+              */
+#ifdef USE_MUSL
+            static constexpr uint32_t check_frequency = 128;
```

Comment:
Interesting. Is MUSL stack smaller?

Reply:
Yes, the first run made the Fast test crashing, and the Fast test uses Musl (the intention is to migrate to it eventually and keep it more or less supported before that time). I could set the frequency to 128 for a normal build as well, but I'm afraid it is too frequent.